### PR TITLE
[ticket/15624] fix forum-image

### PR DESCRIPTION
**Pull request by hanakin**
PHPBB3-15624

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15624

before: 

screenshot 2018-04-10 17 51 22

after: 

screenshot 2018-04-10 17 51 13

### CODE REVIEW

File: phpBB/styles/prosilver/template/forumlist_body.html [hunk 1]
```diff
@@ -44,7 +44,7 @@
 						<!-- ENDIF -->
 						<!-- IF forumrow.FORUM_IMAGE -->
 							<!-- EVENT forumlist_body_forum_image_before -->
-							<span class="forum-image"><!-- EVENT forumlist_body_forum_image_prepend -->{forumrow.FORUM_IMAGE}<!-- EVENT forumlist_body_forum_image_append --></span>
+							<span class="forum-image"><!-- EVENT forumlist_body_forum_image_prepend --><img src="{forumrow.FORUM_IMAGE_SRC}" alt="" width="17px" /><!-- EVENT forumlist_body_forum_image_append --></span>
```
[the remark this answers is not in the document]
Comment (hanakin) [reply phpBB/styles/prosilver/template/forumlist_body.html]: the purpose is i am not sure if we need something there i am looking at that right now

[the remark this answers is not in the document]
Comment (hanakin) [reply phpBB/styles/prosilver/template/forumlist_body.html]: currently it uses the same one as the forum-imageset

https://www.phpbb.com/customise/db/style/pycode/demo/3.2

which i do not think is valid...

**Comment by 3D-I** [comment]
I'd remove also the `float: left;` from the css `.forum-image` in this case.

**Comment by hanakin** [comment]
no float left is still required to force the alignment, i could be changed to inline but thats pointless at this point for prosilver

File: phpBB/styles/prosilver/template/forumlist_body.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (vinny) [reply phpBB/styles/prosilver/template/forumlist_body.html]: I suggest using the forum name. 

`alt="{forumrow.FORUM_NAME}"`

[the remark this answers is not in the document]
Comment (vinny) [reply phpBB/styles/prosilver/template/forumlist_body.html]: In addition, use `width="17px"` it's unnecessary. The administrator should be free to use the desired dimension in their images.

[the remark this answers is not in the document]
Comment (hanakin) [reply phpBB/styles/prosilver/template/forumlist_body.html]: no! 17px is required for retina. as for `L_FORUM_IMAGE` does that even have a value its not in the code

**Comment by 3D-I** [comment]
@hanakin 

With the inline style `width="17px"` the `float: left;` in the css `.forum-image` seems redundant to me, I did test it with FF's inspector.

The lang bit `L_FORUM_IMAGE` reveals this lang key `'FORUM_IMAGE'	=> 'Forum image',` located in `language/en/acp/forums.php`. Not really a concrete value IMO.

**Comment by hanakin** [comment]
The float is required I tested it as well

**Comment by hanakin** [comment]
so looking into it i am going to remove the alt tag for 3.2 if anything this should be a configured field in the db/acp down the road perhaps in 3.3

**Comment by Solidjeuh** [comment]
I made this fix in 3.2.3RC.1, but those images are WAY to small ..

**Comment by hanakin** [comment]
@Solidjeuh not sure what you are saying? if your icon needs to be larger change the size of the icon in the code! these are retina images which require a hard size to be half the size of the source.

**Comment by DavidIQ** [comment]
@hanakin it should be he other way around. We shouldn’t be setting an arbitrary size for the image. If a particular style needs to do that then they should edit the code and hard code the size as has been done here.

**Comment by hanakin** [comment]
@davidiq we have to hardcore it on the img the reason I am doing it via html rather than css is because I would need to create a new class on the img. If that is considered the better option for BC then fine but either way a hard size needs to be specified because we are using an src and not a background img

**Comment by DavidIQ** [comment]
I guess I'm not understanding why this needs to be done at all if this is only broken in the style you posted your screenshot of. Certainly isn't an issue for prosilver, or hasn't been reported as an issue for prosilver.

**Comment by Solidjeuh** [comment]
Hard coded it is fine, but why 17px ? I have my forum images set to 32 x 32px, and it looks pretty good like that.

**Comment by hanakin** [comment]
That’s fine for your board but it’s not what that image is meant to be it is a 17px image that was meant to be aligned to the topic link.

**Comment by marc1706** [comment]
Looking at this, it seems users are using this as a secondary icon next to the forum icon (which is therefore somewhat expected to be 32px). @hanakin thoughts on this?
FWIW; I'll postpone this to 3.2.4.

**Comment by hanakin** [comment]
@marc1706 ultimately it's not a feature I will support in this manner in the new theme, It will be one or the other and as you stated set size of 40px... but for prosilver I would just suggest we close it. Users seem to want to for some reason edit the sizes of each one independently?

**Comment by Noxwizard** [comment]
@marc1706 They don't even expect it to be 32px. They expect it to be whatever size the image they use is. 

I'm not sure why there's a push for it to be a restricted size since in the 10 years that phpBB3 has been out, I've never seen a board use such a small image for a forum icon. A lot of times it's random size images that they've found online. If this gets rolled into prosilver, I know the support forum will see a lot of users asking why their images are distorted now and how to "fix" it.

**Comment by hanakin** [comment]
@Noxwizard

As I stated I will leave it as is for prosilver for now. The problem is it’s not responsive or retina and we are transitioning everything to both. It can be handled in a clean way. It the size needs to be barcoded an uniform.

**Comment by marc1706** [comment]
Ok, will close this for now then.

**Pull request closed without merging**